x/subscription/keeper/msg_server_auto_renewal.go

```diff
@@ -0,0 +1,33 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/x/subscription/types"
+)
+
+func (k msgServer) AutoRenewal(goCtx context.Context, msg *types.MsgAutoRenewal) (*types.MsgAutoRenewalResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	sub, found := k.GetSubscription(ctx, msg.Creator)
+	if !found {
+		return nil, utils.LavaFormatWarning("could not change auto-renewal of subscription", fmt.Errorf("subscription not found"),
+			utils.Attribute{Key: "sub_consumer", Value: msg.Creator},
+		)
+	}
+
+	sub.AutoRenewal = msg.Enable
+	err := k.subsFS.AppendEntry(ctx, msg.Creator, uint64(ctx.BlockHeight()), &sub)
```

Comment:
this should be at the same block as the original subscription, we dont want another instance in the fixation

Reply:
done. see https://github.com/lavanet/lava/commit/b9acd69c93fc42fce0e164674af4de9c33525130